### ISSUE TITLE
tf.distribute.MirroredStrategy makes wrong outputs with tf.keras.layers.Dense

### ISSUE THREAD

**Issue by Ssojux2** [tensorflow/tensorflow#53088]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution : Linux Ubuntu 18.04
- TensorFlow installed from binary:
- TensorFlow version: 2.6.2
- Python version: 3.8
- CUDA/cuDNN version: cuda 11.3,  8.2.1
- GPU model and memory:  RTX 5000, 2 EA, 16gb

when I'm using mirroredstrategy, the model makes wrong outputs.

![image](https://user-images.githubusercontent.com/9550691/142134915-29371557-c661-4856-9f35-fdb70976ca36.png)

this is logs and the code is like

![image](https://user-images.githubusercontent.com/9550691/142134955-8adc2634-9f9e-483b-b151-065643611097.png)

it's part of music transformer.

the logits changes when I using 2 gpus.
I don't no why it make wrong output shape.

there is no problem with 1 gpu.

I don't no why it works like this.

**Comment by Ssojux2** [comment]
Oh I think I make mistakes. subcalssing code is wrong
